ovn-tester/cms/ovn_kubernetes/__init__.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2023 Canonical
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from collections import namedtuple
+
+from ovn_context import Context
+from ovn_workload import WorkerNode, Cluster
+from ovn_utils import DualStackSubnet
+
+ClusterBringupCfg = namedtuple("ClusterBringupCfg", ["n_pods_per_node"])
+
+
+class ovn_kubernetes(object):
+    def __init__(self, cms_config):
+        self.read_config(cms_config)
+
+    def read_config(self, cms_config):
+        bringup_args = cms_config.get("base_cluster_bringup", dict())
+        self.bringup_cfg = ClusterBringupCfg(
+            n_pods_per_node=bringup_args.get("n_pods_per_node", 10)
+        )
+
+    def create_nodes(self, cluster_config, workers):
+        mgmt_net = cluster_config.node_net
+        mgmt_ip = mgmt_net.ip + 2
+        internal_net = cluster_config.internal_net
+        external_net = cluster_config.external_net
+        gw_net = cluster_config.gw_net
+        worker_nodes = [
+            WorkerNode(
+                workers[i % len(workers)],
+                f"ovn-scale-{i}",
+                mgmt_net,
+                mgmt_ip + i,
+                DualStackSubnet.next(internal_net, i),
+                DualStackSubnet.next(external_net, i),
+                gw_net,
+                i,
+            )
+            for i in range(cluster_config.n_workers)
+        ]
+        return worker_nodes
+
+    def prepare_test(self, central_node, worker_nodes, cluster_cfg, brex_cfg):
```

Comment:
Would it make sense to move that function to `ovn-tester/ovn_tester.py`? It looks rather generic to me

Reply:
It does indeed look like one of the generic ones, I think my thinking at the time was that `Cluster` potentially is CMS specific. I'll consider moving it back. Thanks @felixhuettner !

Reply:
In #175 this does become CMS specific:
https://github.com/ovn-org/ovn-heater/pull/175/commits/c964fb20787aa7ad430623a6ad4db755a99e8e1d#diff-b2472ea71d24c06480c4a2d181759380d75d19cd409936937d72927de0fe80c5R146